### PR TITLE
Add nginx_wrk benchmark

### PR DESCRIPTION
**Pull request by meteorfox**
Adds the nginx_wrk benchmark.

Includes unit test as well

### CODE REVIEW

File: benchpress/install_nginx.sh [hunk 1]
```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+set -e
+set -x
+
+NGINX_VERSION='nginx-1.12.2'
+NGINX_TAR_FILE="${NGINX_VERSION}.tar.gz"
+NGINX_DOWNLOAD_URL="http://nginx.org/download/${NGINX_TAR_FILE}"
```
Comment (adhanotia) [inline benchpress/install_nginx.sh:7]: Can we install a prebuilt version of nginx using yum? I see a few nginx versions available for CentOS, eg - https://www.cyberciti.biz/faq/how-to-install-and-use-nginx-on-centos-7-rhel-7/

File: benchpress/install_nginx.sh [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+PCRE_VERSION='pcre-8.41'
+PCRE_TAR_FILE="${PCRE_VERSION}.tar.gz"
+PCRE_DOWNLOAD_URL="ftp://ftp.csx.cam.ac.uk/pub/software/programming/pcre/${PCRE_TAR_FILE}"
```
Comment (adhanotia) [inline benchpress/install_nginx.sh:11]: Same thing, install pcre using yum install?

File: benchpress/jobs/jobs.yml [hunk 3]
```diff
@@ -71,4 +71,11 @@
     num-threads: 1
     scale-factor: 1
     runtime: 1
-
+- benchmark: nginx_wrk_bench
+  name: nginx_wrk_bench default
+  description: defaults for nginx_wrk_bench
+  args:
+    - '-c100'  # Connections
```
Comment (adhanotia) [inline benchpress/jobs/jobs.yml]: Can we keep the default inputs to match what I had in the wrk run script?

File: benchpress/templates/nginx.conf [hunk 4]
```diff
@@ -0,0 +1,117 @@
+
+#user  nobody;
+worker_processes  1;
```
Comment (adhanotia) [inline benchpress/templates/nginx.conf:3]: The default here might need to be changed as well. Look up in the install script?

File: benchpress/tests/test_nginx_wrk_parser.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+            '50%   62.12ms',
+            '75%   63.36ms',
+            '90%   74.62ms',
+            '99%    1.27s',
+            '15545 requests in 10.01s, 12.60MB read',
+            'Socket errors: connect 0, read 0, write 0, timeout 25',
+            'Requests/sec:   1552.47',
+            'Transfer/sec:      1.26MB',
+        ]
+        metrics = self.parser.parse(output, None, 0)
```
Comment (adhanotia) [inline benchpress/tests/test_nginx_wrk_parser.py:35]: If this runs on some other machine, we will get different requests/sec. How does the hard coding work? Would it help to just have the parser check if all output fields exist and not check the actual results?

**Review by adhanotia — CHANGES_REQUESTED** [review]

File: benchpress/tests/test_nginx_wrk_parser.py (hunk 5 above)
Comment (meteorfox) [reply benchpress/tests/test_nginx_wrk_parser.py:35]: This output is hard-coded,  this is the unit test for the parser. This doesn't run the benchmark itself.

This is to make sure the parser works correctly.

File: benchpress/templates/nginx.conf (hunk 4 above)
Comment (meteorfox) [reply benchpress/templates/nginx.conf:3]: I was looking at the benchmark scripts you sent me a while back, I couldn't find conf settings for nginx itself, except for a comment to change the port other than port 80.

But, if you have suggestion let me know.

File: benchpress/jobs/jobs.yml (hunk 3 above)
Comment (meteorfox) [reply benchpress/jobs/jobs.yml]: Sure, I'll update it. 👍

File: benchpress/install_nginx.sh (hunk 2 above)
Comment (meteorfox) [reply benchpress/install_nginx.sh:11]: See  my comment above.

File: benchpress/install_nginx.sh (hunk 1 above)
Comment (meteorfox) [reply benchpress/install_nginx.sh:7]: Sure, we can do that.

But the reasons I went with building from source are:
- No need for root privileges to install or run the benchmark
- Pinned version of the source code so we can reproduce the results.
- It doesn't depend on package installers

**Review by adhanotia — APPROVED** [review]
Awesome!

**Pull request merged**